js/style/style_layer.js

```diff
@@ -114,15 +156,10 @@ StyleLayer.prototype = {
         }
     },
 
-    getPaintValue: function(name, zoom, zoomHistory) {
-        return this._transitions[name].at(zoom, zoomHistory);
-    },
-
     // update zoom
     recalculate: function(zoom, zoomHistory) {
-        this.paint = new PaintProperties[this.type]();
-
-        for (var name in this._transitions) {
+        this.paint = {};
+        for (var name in this._paintSpecifications) {
```

Comment:
The advantage of PaintProperties/LayoutProperties class was that default values were inherited through prototype so you didn't have to loop through them when recalculating styles. So this change can potentially affect performance. 

BTW, Instead of running the fps benchmark, I'd suggest running profiler and seeing how much of a frame CPU time is taken by recalculation.